tests/regression/src/lib.rs

```diff
@@ -210,11 +212,18 @@ mod tests {
 
     struct DiffProfile {
         test_name: String,
+        prev_profile_count: i64,
     }
     impl DiffProfile {
         fn new(prev_profile: &RawProfile, curr_profile: &RawProfile) -> Self {
             let diff_profile = Self {
                 test_name: curr_profile.test_name.clone(),
+                // reads the annotated profile for previous instruction count
+                prev_profile_count: find_instruction_count(
```

Comment:
Hmm, I guess this works but I think it's making some assumptions that we could make more explicit in the flow.

1. Make DiffProfile take in `prev_annotated_profile: &AnnotatedProfile`. There is a requirement that the annotated profile has been constructed for us to successfully diff, and this would make that explicit.
2. Then we could make `instruction_count() -> u64` a member on AnnotatedProfile, which I think would look a little neater.

Reply:
Denoting a "previous AnnotatedProfile" would require querying on annotated profiles on top of the query for raw profiles (required to cg --diff). Instead, `instruction_count()->u64` could be a member of RawProfile which would address those assumptions without adding the extra complexity of querying twice.

Reply:
Maybe there could be an `associated_annotated_profile()` method which would return the `AnnotatedProfile` associated with the `RawProfile`? Because you are assuming that the `AnnotatedProfile` has already been created? Or something like that.

```
prev_profile.get_annotated_profile().instruction_count();
```

I'm not particularly commited to that API, but the current approach seems to be just sneakily querying the `annotated` profile (with the `replace(.raw, .annotated)`)

Reply:
That works. But, for the .get_annotated_profile() function, the `replace(.raw, .annotated)` might still be needed. Otherwise, the alternatives I see are 

1. Adding a `Option <AnnotatedProfile>` field and a `set_annotated_profile(&self, AnnotatedProfile)` to RawProfile which sets that field after an AnnotatedProfile is instantiated in `AnnotatedProfile::new(raw_profile: &RawProfile)` by calling `raw_profile.set_annotated_profile(annotated_profile.clone());`. But the clone requirement could be costly if the AnnotatedProfile becomes larger and this approach would implicitly link Raw and AnnotatedProfile which adds some complexity to the ownership model for future changes. 

2. Another option could be to explicitly link by calling set_annotated_profile after a new AnnotatedProfile has been created in `run_test_mode`. This would get around the implicit linking and cloning issues but would still have linked references between Raw and Annotated which, again, complicate the ownership model. 

With that said the .replace() approach may be even less extensible since any change to the file storage scheme or naming convention would require overhauling it, so I'd prefer option number 2 with explicit linking.

Reply:
Okay, after some offline discussion it makes sense to create a new annotated_profile in that get_annotated_profile which relies on the same test_name, commit_id as the raw profile. That annotated_profile can then be used to derive the instruction count which would get us around the issue of overhauling this when there are storage schema/filename changes without changing the ownership model 😁